yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -238,6 +257,33 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  /** Get a list of preemption containers through AM-RM heartbeat. */
+  private def preemptResources(allocateResponse: AllocateResponse): Unit = {
+    val preemptMessageOpt = Option(allocateResponse.getPreemptionMessage)
```

Comment:
Part of me things these `Option` instances are not necessary and good old null checks would be just as readable, but I don't really mind either way.